cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -0,0 +1,192 @@
+""" Code to allow module store to interface with courseware index """
+from __future__ import absolute_import
+
+from datetime import timedelta
+import logging
+
+from django.conf import settings
+from django.utils.translation import ugettext as _
+from eventtracking import tracker
+from xmodule.modulestore import ModuleStoreEnum
+from search.search_engine_base import SearchEngine
+
+
+# Use default index and document names for now
+INDEX_NAME = "courseware_index"
+DOCUMENT_TYPE = "courseware_content"
+REINDEX_AGE = timedelta(0, 60)
+
+log = logging.getLogger('edx.modulestore')
+
+
+def indexing_is_enabled():
+    """
+    Checks to see if the indexing feature is enabled
+    """
+    return settings.FEATURES.get('ENABLE_COURSEWARE_INDEX', False)
+
+
+class SearchIndexingError(Exception):
+    """ Indicates some error(s) occured during indexing """
+
+    def __init__(self, message, error_list):
+        super(SearchIndexingError, self).__init__(message)
+        self.error_list = error_list
+
+
+class CoursewareSearchIndexer(object):
```

Comment:
@martynjames one minor suggestion (and sorry I didn't brought this in earlier) - could you please convert it to normal class with instance or class methods? If you noticed I've done exactly so in https://github.com/edx/edx-platform/pull/7448 - libraries are indexed a little bit different, so (1) correct indexer need to be picked up based on what we are indexing and (2) most of the code in this class can be reused, but with some overrides. So if you could convert this to classmethods it would be a bit simpler for me.